aibtc-v1/crews/contract_code_generator.py

```diff
@@ -0,0 +1,194 @@
+        print(f"Added new contract: {contract_name}")
+
+        # Write the contract code to the contract file
+        contract_file_path = os.path.join("contracts", f"{contract_name}.clar")
+        with open(contract_file_path, "w") as contract_file:
+            contract_file.write(contract_code)
+        print(f"Wrote code to {contract_file_path}")
+
+        # Check the syntax of the contract
+        result = subprocess.run(
+            ["clarinet", "check", contract_name], capture_output=True, text=True
+        )
+        print(f"Syntax check result: {result.stdout}")
+
+        return f"Successfully created project '{project_name}', added contract '{contract_name}', and checked its syntax."
+    except subprocess.CalledProcessError as e:
+        return f"Error: {e}"
+    finally:
+        os.chdir(initial_dir)
+
+
+def get_llm(model_name):
+    if model_name.startswith("gpt"):
+        return ChatOpenAI(model_name=model_name)
+    elif model_name == "llama2":
+        return OllamaLLM(model=model_name)
+    else:
+        raise ValueError(f"Unsupported model: {model_name}")
+
+
+# Set your desired model name
+model_name = "gpt-4o"  # Example with OpenAI GPT-4
+llm = get_llm(model_name)
+
+# Define the Clarity Code Generator Agent
+clarity_code_generator = Agent(
+    role="Clarity Code Generator",
+    goal="Generate Clarity code for a smart contract on the Bitcoin blockchain based on user input requirements.",
+    verbose=True,
+    memory=True,
+    backstory=(
+        "You are an expert in Clarity, a smart contract language for the Bitcoin blockchain. "
+        "Your goal is to write secure, efficient, and functional Clarity code based on user instructions. Do not create or write anything beside code."
+    ),
+    allow_delegation=False,
+    llm=llm
+)
+
+# Define the Clarity Code Reviewer Agent
+clarity_code_reviewer = Agent(
+    role="Clarity Code Reviewer",
+    goal="Review the generated Clarity code, create a Clarinet project, and check its syntax.",
+    verbose=True,
+    memory=True,
+    backstory=(
+        "You are a meticulous Clarity code reviewer known for ensuring smart contract security and code quality. "
+        "Your goal is to analyze the generated code, create a Clarinet project, and check the syntax."
+    ),
+    allow_delegation=False,
+    llm=llm,
+    tools=[runClarinet]
+)
+
+# Define the Clarity Code Compiler Agent
+clarity_code_compiler = Agent(
+    role="Clarity Code Compiler",
+    goal="Combine the generated Clarity code and the code review report into a comprehensive output for the Streamlit app.",
+    verbose=True,
+    memory=True,
+    backstory=(
+        "You are a Clarity code compilation expert responsible for taking the generated Clarity code and the code review report, and combining them into a final, comprehensive output that can be displayed in the Streamlit app."
+    ),
+    allow_delegation=False,
+    llm=llm
+)
+
+# Define the task for generating Clarity code
```

Comment:
Could this be linked to user input in the streamlit UI rather than a pre-defined task? That way the user can request what they want but the same loop is executed.